server/web/router.go

```diff
@@ -823,6 +818,13 @@ func (p *ControllerRegister) serveHttp(ctx *beecontext.Context) {
 			}
 		}
 	}
+	ctx.Input.ExecMethod = runMethod
+	ctx.Input.ExecController = runRouter
+
+	// execute middleware filters
+	if len(p.filters[BeforeExec]) > 0 && p.execFilter(ctx, urlPath, BeforeExec) {
```

Comment:
you could not move this block here. `BeforeExec` means that we must invoke these filters before `serveHttp`